File: Level 1/문자열 나누기/solution.js
function solution(s) {
  let firstChar = "";
  let sameCount = 0;
  let diffCount = 0;

  let answer = 0;

  [...s].forEach((char) => {
    if (!firstChar) {
      firstChar = char;
      sameCount++;
      answer++;
      return;
    }

    char === firstChar ? sameCount++ : diffCount++;

    if (sameCount === diffCount) {
      firstChar = "";
      sameCount = 0;
      diffCount = 0;
    }
  });

  return answer;
}

File: Level 1/문자열 나누기/solution.py
def solution(s):
    first_char = ""
    same_count = 0
    diff_count = 0

    answer = 0

    for char in s:
        if not first_char:
            first_char = char
            same_count += 1
            answer += 1
            continue

        if char == first_char:
            same_count += 1
        else:
            diff_count += 1

        if same_count == diff_count:
            first_char = ""
            same_count = 0
            diff_count = 0

    return answer
